kedro/framework/cli/jupyter.py

```diff
@@ -31,20 +31,56 @@
 OVERWRITE_HELP = """If Python file already exists for the equivalent notebook,
 overwrite its contents."""
 
+jupyter_command_group_order = ["init", "notebook", "lab", "convert"]
+jupyter_command_group_order = {
+    command: order for order, command in enumerate(jupyter_command_group_order)
+}
+
+
+class JupyterCommandGroup(click.Group):
```

Comment:
Love the idea of doing this 💯 But why doesn't `list_commands` just do `return ["init", "notebook", "lab", "convert"]`? I get that `self.commands` is unordered, but then `list_commands` is surely just returning the original ordered list here? 🤔 

Not a big thing, but I feel like there's a more direct way of getting the correctly ordered commands here.

Reply:
(Y) This actually works. As what I get from `self.commands` is `Dict[str|Commands]` . I didn't notice it actually works with a list of string and it would resolve the commands. Brilliant.

Reply:
Excellent! Do we even need to override `__init__` now or just `list_commands`?